src/main/java/seedu/address/model/module/Code.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.model.module;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Module's code in FaculType.
+ * Guarantees: immutable; is valid as declared in {@link #isValidCode(String)}
+ */
+public class Code {
```

Comment:
Perhaps it is better to change this into ModuleCode instead?

Reply:
I think it is a good idea, it is more descriptive (and has the same format with ModuleName)